src/parse_input.py

```diff
@@ -0,0 +1,153 @@
+                                                        archetypes_in_input,
+                                                        commodity_dictionary)
+
+    return facility_dictionary
+
+
+def get_facility_and_commod_names(root, archetypes_in_input,
+                                  commodity_dictionary):
+    '''
+    facility_dict = get_facility_and_commod_names(root, archetypes_in_input,
+                                    commodity_dictionary)
+
+    get_facility_and_commodity_names reads a Cyclus input file element tree and
+    finds the facility names for each facility specified in a <facility> tag.
+    Then, using the facility archetype and module, the input and output
+    commodities are retrieved.
+
+    inputs:
+        - root: an xml.etree.ElementTree root of a Cyclus input file tree
+        - archetypes_in_input: a dictionary with format
+                               {'archetype' : 'module::archetype'}
+        - commodity_dictionary: a dictionary with the Cyclus archetypes
+        available and the archetype tags of their incommodities and
+        outcommodities
+
+    outputs:
+        - facility_dict: a dictionary of each facility and its
+        incommodities and outcommodities. format:
+        {'facility' : (['incommodities'], ['outcommodities'])}
+    '''
+
+    facility_dict = {}
+
+    for facility in root.findall('./facility'):
+        facility_name = facility.find('name').text
+        facility_archetype = facility.find('config/').tag
+        facility_module = archetypes_in_input[facility_archetype]
+        commodities = commodity_dictionary[facility_module]
+
+        in_commod_tags = commodities[0]
+        out_commod_tags = commodities[1]
+
+        facility_in_commods = []
+        facility_out_commods = []
+
+        for archetype_tag in facility.find('.config/' + facility_archetype):
+
+            in_commods = find_commod(archetype_tag, in_commod_tags)
+            if in_commods is not None:
+                facility_in_commods.extend(in_commods)
+
+            out_commods = find_commod(archetype_tag, out_commod_tags)
+            if out_commods is not None:
+                facility_out_commods.extend(out_commods)
+
+        facility_dict[facility_name] = (facility_in_commods,
+                                        facility_out_commods)
+
+    return facility_dict
+
+
+def archetypes_in_input_file(root):
+    '''
+    input_archetypes = archetypes_in_input_file(root)
+
+    archetypes_in_input_file reads a Cyclus input file element tree and finds
+    the modules and archetypes that are defined in the <archetypes> tag.
+
+    inputs:
+        - root: an xml.etree.ElementTree root of a Cyclus input file tree
+
+    outputs:
+        - archetypes_in_input: a dictionary with format
+                               {'archetype' : 'module::archetype'}
+    '''
+
+    archetypes_in_input = {}
+    for archetypes in root.findall('./archetypes/'):
```

Comment:
Singular is probably more appropriate here
```suggestion
    for archetype in root.findall('./archetypes/'):
```